pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -44,13 +43,30 @@ func (p *Probe) Check() error {
 
 	// Envoy has received some configuration, make sure that configuration has been received for
 	// all inbound ports.
-	if err := p.checkInboundConfigured(); err != nil {
+	if err := p.checkInbound(); err != nil {
 		return err
 	}
 
 	return p.checkServerInfo()
 }
 
+func (p *Probe) checkInbound() error {
+	if p.NodeType != model.SidecarProxy {
```

Comment:
I'm not up to speed with the recent developments WRT listeners. Why do you skip this for sidecars?

Reply:
The background is 
- Check() includes `checkInbound()` and `checkServerInfo()`
- We are deprecating the `containerPorts` and use single 15006 listener for inbound traffic

checkServerInfo() could pass only if the envoy initialization is done, which will pass once the inbound listener(15001 and 15006) is ready

If 15001 and 15006 are always provided by pilot, the followed `checkServerInfo` covers checkInboundConfigued.

Reply:
@costinm has the fix for that